src/cpu/gemm_convolution.hpp

```
@@ -71,7 +75,30 @@ struct gemm_convolution_fwd_t : public primitive_t {
                 bool ok = true;
 
                 for (int i = 0; i < po.len(); i++) {
-                    ok = ok && utils::one_of(po.entry_[i].kind, sum, binary, eltwise, depthwise, quantization);
+                    const auto &post_op = po.entry_[i];
+#if DNNL_X64
+                    using namespace cpu::x64;
+                    cpu_isa_t isa = isa_undef;
+                    if (po.entry_[i].kind == binary) {
+                        auto dst_md = this->dst_md();
+                        if (mayiuse(avx512_core))
+                            isa = avx512_core;
+                        else if (mayiuse(avx2))
+                            isa = avx2;
+                        else if (mayiuse(sse41))
+                            isa = sse41;
+                        if ((isa == isa_undef)
+                                || !binary_injector::is_supported(isa,
+                                        binary_injector::get_src1_desc(
+                                                post_op, *dst_md),
+                                        *dst_md, default_strategies())) {
+                            return false;
+                        }
+                    }
+#endif
+                    ok = ok
+                            && utils::one_of(post_op.kind, sum, binary, eltwise,
+                                    depthwise, quantization);
```

It's more logical to move this line before the specific binary post op checks. Just to fail fast.

---

have moved this post_op.kind check before specific binary checks